spring-pulsar/src/test/java/org/springframework/pulsar/core/DefaultTopicResolverTests.java

```diff
@@ -44,6 +50,8 @@ class DefaultTopicResolverTests {
 
 	private static final String fooTopic = "foo-topic1";
 
+	private static final String bazTopic = "baz-topic1";
```

Comment:
By the time the topic comes through here it would already be resolved. SpEL can be used on `@PulsarListener(topics = ...` and property placeholders in Boot config properties under `spring.pulsar.producer|consumer|reader.topic*`.

Reply:
Think you misunderstood me. I meant something like this `@PulsarMessage(topic = "${my-app.topics.topic-alias}")`

I don't see why this can't be added by modifying the `getAnnotatedTopicInfo` like so:

```java
String getAnnotatedTopicInfo(Class<?> messageType) {
return this.pulsarMessageAnnotationRegistry.getAnnotationFor(messageType)
	.map(PulsarMessage::topic)
        .map(Environment::getProperty)
	.filter(StringUtils::hasText)
	.orElse(null);
}
```

This way one can reuse the property in the `@PulsarListener` and `@PulsarMessage`

Reply:
Ahh yes. Got it. This is the same thing we allow users to do in `@PulsarListener` and we should do the same thing here. It is not much to add. We can allow SpEL (not just property placeholders). Great idea. I may add that as a subsequent PR as this has multiple moving pieces already.

Reply:
I'd be happy to drop that PR after this one gets merged in.

Reply:
All your's my friend -> https://github.com/spring-projects/spring-pulsar/issues/568